File: TOKI-Learning/pascal/cek_bilangan_prima.pas
var
	n, i, angka, j : longint;
	prima : boolean;
begin
	readln(n);
	for i := 1 to n do
	begin
		readln(angka);
		prima := true;
		for j := 2 to trunc(sqrt(angka)) do
			if angka mod j = 0 then
			begin
				prima := false;
				break;
			end;
		if angka = 1 then
			writeln('BUKAN')
		else if prima then
			writeln('YA')
		else
			writeln('BUKAN');
	end;
end.

File: TOKI-Learning/c++/cek_bilangan_prima.cpp
#include <cstdio>
#include <cmath>

int main()
{
    int n, bi;
    scanf("%d", &n);
    for (int i = 0; i < n; i++)
    {
        int j = 0;
        int prima = 1;
        scanf("%d", &bi);
        for (j = 2; j <= trunc(sqrt(bi)); j++)
            if (bi % j == 0)
            {
                prima = 0;
                break;
            }
        if (bi == 1)
            printf("BUKAN\n");
        else if (prima == 1)
            printf("YA\n");
        else
            printf("BUKAN\n");
    }
    return 0;
}
